### PR TITLE
Deprecate importing `inject` from `@ember/service`

### LINKED ISSUE #20262: Deprecate `inject` export from `@ember/service`
Since [Ember v4.1.0](https://github.com/emberjs/ember.js/blob/master/CHANGELOG.md#v410-december-28-2021), it's possible to use the `service` export coming from `@ember/service`.
This feature was suggested in [RFC 752](https://github.com/emberjs/rfcs/pull/752) and implemented in [PR 19776](https://github.com/emberjs/ember.js/pull/19776).
This means that, at some point, we can deprecate `inject` in favor of `service` as having two things to accomplish the same thing is redundant. This issue serves as a reminder to implement that deprecation.

### PR DESCRIPTION
**Pull request by bertdeblock**
Closes #20262.

### CODE REVIEW

File: packages/@ember/service/index.ts [hunk 1]
```diff
@@ -24,6 +25,21 @@ export function inject(): PropertyDecorator;
...
   ...args: [] | [name: string] | ElementDescriptor
 ): PropertyDecorator | DecoratorPropertyDescriptor | void {
+  deprecate(
+    'Importing `inject` from `@ember/service` is deprecated. Please import `service` instead.',
+    false,
+    {
+      for: 'ember-source',
+      id: 'importing-inject-from-ember-service',
+      since: {
+        available: '5.4.0',
```
Comment (bertdeblock) [inline packages/@ember/service/index.ts:35]: Unsure about these.

File: packages/@ember/service/index.ts [hunk 2]
```diff
@@ -24,6 +25,21 @@ export function inject(): PropertyDecorator;
...
+        enabled: '5.4.0',
+      },
+      until: '6.0.0',
+      url: 'TODO',
```
Comment (bertdeblock) [inline packages/@ember/service/index.ts:39]: https://github.com/ember-learn/deprecation-app/pull/1336

File: packages/@ember/-internals/deprecations/index.ts [hunk 3]
```diff
@@ -139,6 +139,16 @@ export const DEPRECATIONS = {
...
   }),
+  DEPRECATE_IMPORT_INJECT: deprecation({
+    for: 'ember-source',
+    id: 'importing-inject-from-ember-service',
+    since: {
+      available: '6.3.0',
+      enabled: '6.3.0',
+    },
+    until: '7.0.0',
+    url: 'TODO',
```
Comment (bertdeblock) [inline packages/@ember/-internals/deprecations/index.ts:150]: https://github.com/ember-learn/deprecation-app/pull/1336

File: packages/@ember/service/tests/service_test.js [hunk 4]
```diff
@@ -1,47 +1,51 @@
 import Service, { inject, service } from '@ember/service';
 import EmberObject from '@ember/object';
 import { buildOwner, runDestroy } from 'internal-test-helpers';
-import { moduleFor, AbstractTestCase } from 'internal-test-helpers';
+import { moduleFor, AbstractTestCase, expectDeprecation } from 'internal-test-helpers';
 
 moduleFor(
   'inject - decorator',
   class extends AbstractTestCase {
     ['@test works with native decorators'](assert) {
```
Comment (kategengler) [inline packages/@ember/service/tests/service_test.js:9]: I think you need `testUnless` for these tests, ala https://github.com/emberjs/ember.js/blob/3a1806a1a7d1db007f72e452175294479a6e8113/packages/@ember/-internals/glimmer/tests/integration/application/actions-test.js#L61

Comment (bertdeblock) [reply packages/@ember/service/tests/service_test.js]: Thanks!

**Comment by ijlee2** [comment]
If you'd like, you can update [the deprecation guide](https://deprecations.emberjs.com/id/importing-inject-from-ember-service) to list [`ember-codemod-remove-inject-as-service`](https://github.com/ijlee2/ember-codemod-remove-inject-as-service).

**Pull request merged**